rfcs/NOCK-001.md

```diff
@@ -0,0 +1,44 @@
+# Analysis of lifecycle methods
+
+![](https://img.shields.io/badge/rfc-NOCK--001-blue.svg)
+![](https://img.shields.io/badge/status-draft-orange.svg)
```

Comment:
We could also leave this as a "draft" for a bit to let it bake for a while before we accept it. Or go ahead and accept it now. I don't feel strongly either way. If there are no objections let's merge it as a draft.

Reply:
+1 or draft